### ISSUE TITLE
BC PR C: train_bc_cartpole example + slow cloned-vs-expert eval test

### ISSUE THREAD

**Issue by rjwalters** [rjwalters/thrust#169]
**Part of #161** (BC epic). **Depends on #167** (BcTrainer + loss). Labeled `loom:blocked` until #167 merges.

## Background
With `BcTrainer`, `BcConfig`, and the `Demonstrations` dataset in place, this PR delivers the end-to-end Behavioral Cloning story on CartPole: harvest expert demos from a trained A2C policy, clone with BC, and report cloned-vs-expert reward. Per epic decisions 1, 4, and 5 (#161), demos come from rolling out a trained A2C `MlpBurnPolicy` greedily on a `CartPole` `EnvPool` (lowest-friction, fully in-tree, seeded), and the eval follows the A2C two-tier convention (fast always-on smoke already lives in #167; this PR adds the slow `#[ignore]`d convergence/eval test). This is PR C of the BC epic and the deliverable that closes ROADMAP Milestone 6's imitation-learning item.

### 1. `examples/games/cartpole/train_bc_cartpole.rs`
End-to-end pipeline, mirroring the structure/feature-gating of `examples/games/cartpole/train_cartpole_a2c.rs` (`--features training`, optional `wgpu`, seeded):
1. **Expert**: train an A2C policy on CartPole for a fixed budget until it is a strong expert (reuse the A2C rollout+GAE loop from the A2C example).
2. **Harvest**: roll the expert out greedily (argmax over `policy.forward(obs).0` logits) for a fixed number of env steps, collecting `(obs, action)` into a `Demonstrations`.
3. **Clone**: construct a fresh seeded `MlpBurnPolicy` + `BcTrainer`, run `BcConfig::epochs` supervised epochs over the demos, logging per-epoch loss + accuracy.
4. **Report**: roll out BOTH the expert and the cloned policy greedily over N evaluation episodes and print mean episode reward for each (cloned-vs-expert).
Register the example in `Cargo.toml` with an `[[example]]` block (`name = "train_bc_cartpole"`, path under `examples/games/cartpole/`).

### 2. `tests/test_bc_cartpole.rs` — slow gated eval (`#[ignore]`)
Mirror `tests/test_a2c_cartpole.rs` two-tier doc/structure. The slow test (`#[ignore]`, opt-in via `--ignored`):
- Train an A2C expert to the CartPole solved region (or a fixed strong budget), seeded.
- Harvest demos via greedy rollout.
- Clone with `BcTrainer`.
- Roll out the cloned policy and assert **mean episode length >= 150 over N (>= 20) episodes** (random baseline ~22; expert ~195+; 150 cleanly separates "cloned the expert" from "did not learn").

Run with:
```text
cargo test --release --features training --test test_bc_cartpole -- --ignored
```

(The fast always-on smoke test lives in #167's `bc/trainer.rs` `#[cfg(test)]` — this PR does not duplicate it.)

### 3. ROADMAP
Check off `- [ ] Imitation learning (behavioral cloning)` -> `- [x]` in `ROADMAP.md` Milestone 6.

## Acceptance criteria
- `cargo run --example train_bc_cartpole --features training --release` runs end-to-end and prints both cloned and expert mean episode reward.
- `tests/test_bc_cartpole.rs` compiles under `--features training`; the `#[ignore]`d eval passes (cloned mean episode length >= 150) when run with `--ignored --release`.
- Example + test are seeded and reproducible.
- ROADMAP Milestone 6 imitation-learning item is checked off.
- Zero edits to existing trainers.

## Affected files
- NEW `examples/games/cartpole/train_bc_cartpole.rs`
- NEW `tests/test_bc_cartpole.rs`
- EDIT `Cargo.toml` (add `[[example]]` for `train_bc_cartpole`)
- EDIT `ROADMAP.md` (check off imitation-learning item)

## References
- Example structure + EnvPool rollout + GAE + feature gating: `examples/games/cartpole/train_cartpole_a2c.rs`.
- Slow-test two-tier convention + `#[ignore]` gating + seeded `build`/`train` helpers: `tests/test_a2c_cartpole.rs`, `tests/test_sac_pendulum.rs`.
- Greedy action selection from logits: `src/policy/mlp.rs` (`forward` -> argmax over logits).
- ROADMAP line: `ROADMAP.md` Milestone 6, "Imitation learning (behavioral cloning)".
